pkg/operator/ceph/object/user.go

```diff
@@ -37,47 +37,24 @@ const (
 
 // An ObjectUser defines the details of an object store user.
 type ObjectUser struct {
-	UserID      string  `json:"userId"`
-	DisplayName *string `json:"displayName"`
-	Email       *string `json:"email"`
-	AccessKey   *string `json:"accessKey"`
-	SecretKey   *string `json:"secretKey"`
-	SystemUser  bool    `json:"systemuser"`
-}
-
-// ListUsers lists the object pool users.
-func ListUsers(c *Context) ([]string, int, error) {
-	result, err := runAdminCommand(c, true, "user", "list")
-	if err != nil {
-		return nil, RGWErrorUnknown, errors.Wrap(err, "failed to list users")
-	}
-
-	var s []string
-	if err := json.Unmarshal([]byte(result), &s); err != nil {
-		return nil, RGWErrorParse, errors.Wrapf(err, "failed to read users info result=%s", result)
-	}
-
-	return s, RGWErrorNone, nil
-}
-
-type rgwUserInfo struct {
-	UserID      string `json:"user_id"`
-	DisplayName string `json:"display_name"`
-	Email       string `json:"email"`
-	Keys        []struct {
-		AccessKey string `json:"access_key"`
-		SecretKey string `json:"secret_key"`
-	}
+	UserID       string  `json:"userId"`
+	DisplayName  *string `json:"displayName"`
+	Email        *string `json:"email"`
+	AccessKey    *string `json:"accessKey"`
+	SecretKey    *string `json:"secretKey"`
+	SystemUser   bool    `json:"systemuser"`
+	AdminOpsUser bool    `json:"adminopsuser"`
```

Comment:
Can we pass the `Caps` as a `string` in general instead of `bool` specific to adminOpsUser, so that if `cephObjectStoreUser` extend to support `caps` in future we can avoid adding to the new variable `rgwUserInfo` struct.

Reply:
`rgwUserInfo` is gone but it's a good point for `cephObjectStoreUser` to support caps eventually. This struct type `ObjectUser` is only used when calling the `radosgw-admin` which `cephObjectStoreUser` has stopped using with this patch. If caps are implemented they will go in the spec and be reflected as a `admin.User` type. So I'm not sure there is a concern here.

THoughts?